nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/dto/DtoFactory.java

```diff
@@ -2802,6 +2819,24 @@ public Set<ComponentDifferenceDTO> createComponentDifferenceDtosForLocalModifica
        return differencesByComponent.keySet();
    }
 
+   private static final Pattern DIFFERENCE_DESCRIPTION_PATTERN = Pattern.compile(".*artifact='(.*)'.*version='(.*)'].*version='(.*)']'$");
```

Comment:
Introducing this pattern to reformat the description seems potentially brittle. It would be better to either change it elsewhere, or build the description using values available from the `FlowDifference`.

Reply:
I moved this code to the ConciseEvolvingDifferenceDescriptor class where some other difference descriptions are generated, and I do think this made the code cleaner.

Should I rebase this PR onto main, to try to resolve the macos build/test issues in the CI checks?  Or can those checks be ignored for now?

Reply:
Thanks, rebasing would be a good next step.